documentation/modules/managing/con-broker-config-properties.adoc

```diff
@@ -222,10 +242,13 @@ log.retention.ms=1680000
 # ...
 ----
 
-If  `log.retention.ms` is set to -1, no time limit is applied to log retention, so all logs are retained.
-Disk usage should always be monitored, but the -1 setting is not generally recommended as it can lead to issues with full disks, which can be hard to rectify.
+The retention period is based on the time messages were appended to the segment. The milliseconds configuration has priority over minutes, which has priority over hours. 
+The minutes and milliseconds configuration is null by default, but the three options provide a substantial level of control over the data you wish to retain. Preference should be given to the milliseconds configuration, as it is the only one of the three properties that is dynamically updateable.
+
+If `log.retention.ms` is set to -1, no time limit is applied to log retention, and all logs are retained. 
+However, this setting is not generally recommended as it can lead to issues with full disks that are difficult to rectify.
 
-For size-based log retention, you set a maximum log size (of all segments in the log) in bytes:
+For size-based log retention, you specify a maximum log size (in bytes) for all segments in the log:
```

Comment:
It's actually a lower bound. If you set the retention to 1GB, Kafka will ensure that at all time there is at least 1GB of log available. 

For example if `log.retention.bytes` is 1000 and `log.segment.bytes` is 300, Kafka will keep 4 segments + the active segment (so up to 1500 bytes) to ensure at least 1000 bytes are available. Once the active segment fills up and a new segment is rolled, the oldest segment will be deleted and at this point the size on disk will be 1200 (4 full segments + the newly active segment is that empty). So even though the size is set to 1000 bytes, in practice it will vary between 1200 and 1500 bytes (ignoring index files).

Reply:
Thanks. Changed as follows:

> For size-based log retention, you specify a minimum log size (in bytes):
> 
> `log.retention.bytes=1073741824`
> 
> This means that Kafka will ensure there is always at least the specified amount of log data available.
> 
> For example, if you set `log.retention.bytes` to 1000 and `log.segment.bytes` to 300, Kafka will keep 4 segments plus the active segment, ensuring a minimum of 1000 bytes are available. 
> When the active segment becomes full and a new segment is created, the oldest segment is deleted. 
> At this point, the size on disk may exceed the specified 1000 bytes, potentially ranging between 1200 and 1500 bytes (excluding index files).